src/go/k8s/internal/controller/redpanda/cluster_controller.go

```diff
@@ -1023,35 +981,69 @@ func (r *ClusterReconciler) doDecommissionGhostBrokers(c context.Context, vClust
 	if err != nil {
 		return fmt.Errorf("unable to create admin client: %w", err)
 	}
-	adminBrokers, err := adminClient.Brokers(ctx)
+
+	health, err := adminClient.GetHealthOverview(ctx)
+	if err != nil {
+		return fmt.Errorf("feating cluster heatlh overview: %w", err)
+	}
+
+	// Until cluster does not report that its healthy do not decommission
+	if !health.IsHealthy {
```

Comment:
Separate question: Is it possible for the presence of ghost brokers to cause a cluster to be unhealthy?

Reply:
> Do we actually want this to be an error case or should this function just noop if the cluster isn't healthy?

For example, if there's anything after a call to doDecommissionGhostBrokers (Updating .Status comes to mind), returning an error would prevent that from running but I don't think we'd actually want that.

There is only updating License. I could reorder those, so that decommission is the last action in the reconciliation loop.

If this early return should just not decommission anything until cluster become healthy, then I can adjust that too.